extensions/voice/org.eclipse.smarthome.voice.mactts/src/main/java/org/eclipse/smarthome/voice/mactts/internal/MacTTSAudioStream.java

```diff
@@ -74,19 +79,31 @@ private InputStream createInputStream() throws AudioException {
         try {
             Process process = Runtime.getRuntime().exec(command);
             process.waitFor();
-            File file = new File(outputFile);
-            if (file.exists()) {
-                return new FileInputStream(file);
-            } else {
-                throw new AudioException("Temporary file '" + outputFile + "' not found!");
-            }
+            file = new File(outputFile);
+            return getFileInputStream(file);
         } catch (IOException e) {
             throw new AudioException("Error while executing '" + command + "'", e);
         } catch (InterruptedException e) {
             throw new AudioException("The '" + command + "' has been interrupted", e);
         }
     }
 
+    private InputStream getFileInputStream(File file) throws AudioException {
+        if (file == null) {
+            throw new IllegalArgumentException("file must not be null");
+        }
+        if (file.exists()) {
+            this.length = file.length();
```

Comment:
The initialization of length needs to be done only once and not all the time `getClonedStream()` is called.

If we would like to handle file length changes it does not work (IIRC) as the file size could change also after the file input stream has been created.

So, I think this line could be moved above after `file = new File(outputFile);`
Or is it because you don't want to handle FileNotFoundException twice?
But reset the size all the time... Hm

WDYT?